script.js

```diff
@@ -0,0 +1,449 @@
+
+                    // Make a card collapse
+                    collapseButton.addEventListener("click", (e) => {
+                        e.stopPropagation(); // Prevent triggering the card click event
+                        ingredientsList.remove();
+                        recipeDescription.remove();
+                        collapseButton.remove();
+                    });
+        }
+            })
+
+    
+    })
+}
+
+//rendering all recipes
+renderRecipeCard(recipesContainer, allRecipes);
+
+//find recipes by number of ingredients
+const getRecipeByIngredientNbr = (num)=>{
+    const foundRecipes = [];
+    allRecipes.forEach((obj)=>{
+        if(obj.ingredients.length===num) {
+            foundRecipes.push(obj);
+        }
+    })
+    return foundRecipes;
+}
+
+
+ingredientSearchBtn.addEventListener("click", (e)=> {
+    const num = Math.sqrt((parseInt(ingredientNumber.value) ** 2)); // because I'm lazy I am getting the square root of the square of the input, to make sure it's not a negative value
+    result = getRecipeByIngredientNbr(num);
+    renderRecipeCard(recipesSearchResults, result);
+
+})
+
+
+
+// finding recipes by keywords
+const getRecipeByKeyword = (array, str) =>{
+    const foundRecipes = [];
+    //If I understand regex correctly, this is supposed to catch anything that is not a space or a letter
+    const regex = /[^a-zA-Z\s]/g; 
+    const cleanStr = str.replace(regex, "").toLowerCase();
+    //handling empty inputs
+    if (!cleanStr.trim()) { 
+        alert("Please enter a valid keyword.");
+        return;  
+    };
+    array.forEach((obj)=>{
+        let found = false;
+        for (const key in obj) {
+            const value = obj[key];
+            // check for each value if it's a string 
+            if (typeof value === "string" && value.includes(cleanStr)){
+                found = true;
+                break 
+            };
+            // check for the value if it's an array to handle the case of ingredients
+            if ( Array.isArray(value)) {
+                value.forEach((element)=>{
+                    for (const elementKey in element){
+                        const elementValue = element[elementKey]
+                        if (elementValue.includes(cleanStr)){
+                            found = true;
+                            break
+                        }
+                    }
+                })
+            }
+        }
+        if (found) {
+            foundRecipes.push(obj)
+        }
+    });
+    return foundRecipes
+
+}
+
+keywordSearchBtn.addEventListener("click", (e)=>{
+    recipesSearchResults.innerHTML = ""
+    
+    result = getRecipeByKeyword(allRecipes, keywordInput.value);
+    renderRecipeCard(recipesSearchResults, result);
+
+} )
+
+
+//time counter function
+const timeCounter= ()=>{
+    let seconds = 0; 
+    setInterval(()=>{
+    
+        seconds++;
+        timeCount.innerText = seconds;
+        durationUnit.textContent = " seconds";
```

Comment:
what are you trying to do here? 
Another approach you can follow is to have new Date() object to track the date when you started visiting the website and them you compare it to the current date to get the duration (the time difference)